src/allmydata/util/encodingutil.py

```diff
@@ -373,15 +363,8 @@ def listdir_unicode(path):
     Wrapper around listdir() which provides safe access to the convenient
     Unicode API even under platforms that don't provide one natively.
     """
-    precondition(isinstance(path, unicode), path)
-
-    # On Windows and MacOS X, the Unicode API is used
-    # On other platforms (ie. Unix systems), the byte-level API is used
```

Comment:
This part is still true, isn't it? That is, that we _do_ want to call the fallback function on "not Windows, not MacOS"?

Reply:
unicode_platform was defined as `is_unicode_platform = PY3 or sys.platform in ["win32", "darwin"]`

in 53084f76ced048e5516ce1b79d8a3f055eadd389 it was hardcoded as True, now it's gone altogether

Reply:
The function `listdir_unicode_fallback()` can be dropped

Reply:
Ah, okay sounds good -- I was just reading the code/comment didn't look at the wider context, sorry.